src/region/__tests__/RegionCreator-test.tsx

```diff
@@ -12,302 +16,285 @@ describe('RegionCreator', () => {
+    // Render helpers
+    const getWrapper = (props = {}): ReactWrapper => mount(<RegionCreator {...defaults} {...props} />);
+    const getWrapperRoot = (wrapper: ReactWrapper): ReactWrapper => wrapper.find('[data-testid="ba-RegionCreator"]');
+
+    // Layout helpers (defaults to 1000x1000px with no padding)
+    const setWrapperSize = (wrapper: ReactWrapper, x = 0, y = 0, height = 1000, width = 1000): ReactWrapper => {
+        jest.spyOn(getWrapperRoot(wrapper).getDOMNode(), 'getBoundingClientRect').mockImplementation(() =>
+            getDOMRect(x, y, height, width),
+        );
+        return wrapper;
+    };
 
-        wrapper.unmount();
+    beforeEach(() => {
+        jest.useFakeTimers();
 
-        expect(removeListeners).toHaveBeenCalled();
+        jest.spyOn(document, 'addEventListener');
+        jest.spyOn(document, 'removeEventListener');
+        jest.spyOn(window, 'cancelAnimationFrame');
+        jest.spyOn(window, 'requestAnimationFrame').mockImplementation(cb => setTimeout(cb, 100)); // 10 fps
     });
 
-    describe('getPosition()', () => {
-        test('should return the provided coordinate less the left/top position of the containing element', () => {
+    describe('mouse events', () => {
+        const simulateDrawStart = (wrapper: ReactWrapper, clientX = 10, clientY = 10): void =>
+            act(() => {
+                wrapper.simulate('mousedown', { buttons: 1, clientX, clientY });
+            });
+        const simulateDrawMove = (clientX = 10, clientY = 10): void =>
+            act(() => {
+                document.dispatchEvent(new MouseEvent('mousemove', { buttons: 1, clientX, clientY }));
+            });
+        const simulateDrawStop = (): void =>
+            act(() => {
+                document.dispatchEvent(new MouseEvent('mouseup'));
+            });
+
+        test('should start the render loop and add all event listeners when starting', () => {
             const wrapper = getWrapper();
-            const instance = getInstance(wrapper);
-            const creatorRef = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
 
-            instance.creatorRef = { current: creatorRef };
+            simulateDrawStart(wrapper);
+            simulateDrawMove(50, 50);
 
-            jest.spyOn(creatorRef, 'getBoundingClientRect').mockImplementation(() => getDOMRect(15, 15, 50, 50));
+            expect(document.addEventListener).toHaveBeenCalledWith('mousemove', expect.any(Function));
+            expect(document.addEventListener).toHaveBeenCalledWith('mouseup', expect.any(Function));
+            expect(window.requestAnimationFrame).toHaveBeenCalled();
+        });
+
+        test('should update the rendered rect when the user draws', () => {
+            const wrapper = getWrapper();
 
-            expect(instance.getPosition(100, 100)).toEqual([85, 85]);
+            simulateDrawStart(wrapper);
+            simulateDrawMove(50, 50);
```

Comment:
Forgot `expect`?

Reply:
Nope.

Reply:
Whoops, on second look it's just a dead test. I misread the diff even after your comment. Nice catch, will fix.